sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/analysis/AnalysisSuite.scala

```diff
@@ -217,5 +217,23 @@ class AnalysisSuite extends AnalysisTest {
       nullResult,
       udf4)
     // checkUDF(udf4, expected4)
+    }
+
+  test("SPARK-11863 mixture of aliases and real columns in orderby clause - tpcds 19,55,71") {
+    val a = testRelation2.output(0)
+    val c = testRelation2.output(2)
+    val alias1 = a.as("a1")
+    val alias2 = c.as("a2")
+    val alias3 = count(a).as("a3")
+
+    val plan = testRelation2.
+      groupBy('a, 'c) ('a.as("a1"), 'c.as("a2"), count('a).as("a3")).
+      orderBy('a1.asc, 'c.asc)
```

Comment:
@cloud-fan Will do. Wenchen, there are a few test failures. I am still looking at it. So i think our idea to NOT consider the already resolved attribute for pushdown decision is causing the issue.

Here are the tests
1) SELECT count(*) FROM orderByData GROUP BY a ORDER BY count(*)
    In this case we want the sort attribute representing the count(*) to be replace by the
    group by alias.
2) SELECT a FROM orderByData GROUP BY a ORDER BY a, count(*), sum(b)
    In this case we want the count(*) to be pushed down to aggregate

In both these case, we are skipping pushdown processing because its a resolved attribute.
Given this wenchen, may i request you to look at the original fix. After learning more about
different conditions, it seems like that may be a safer fix. Let me know what you think.